src/inlines.c

```diff
@@ -1137,17 +1137,68 @@ static cmark_node *handle_close_bracket(cmark_parser *parser, subject *subj) {
   // What if we're a footnote link?
   if (parser->options & CMARK_OPT_FOOTNOTES &&
       opener->inl_text->next &&
-      opener->inl_text->next->type == CMARK_NODE_TEXT &&
-      !opener->inl_text->next->next) {
+      opener->inl_text->next->type == CMARK_NODE_TEXT) {
+
     cmark_chunk *literal = &opener->inl_text->next->as.literal;
-    if (literal->len > 1 && literal->data[0] == '^') {
-      inl = make_simple(subj->mem, CMARK_NODE_FOOTNOTE_REFERENCE);
-      inl->as.literal = cmark_chunk_dup(literal, 1, literal->len - 1);
-      inl->start_line = inl->end_line = subj->line;
-      inl->start_column = opener->inl_text->start_column;
-      inl->end_column = subj->pos + subj->column_offset + subj->block_offset;
-      cmark_node_insert_before(opener->inl_text, inl);
-      cmark_node_free(opener->inl_text->next);
+
+    // look back to the opening '[', and skip ahead to the next character
+    // if we're looking at a '[^' sequence, and there is other text or nodes
+    // after the ^, let's call it a footnote reference.
+    if (literal->data[0] == '^' && (literal->len > 1 || opener->inl_text->next->next)) {
```

Comment:
This changeset tackles footnote reference labels that contain the letter `w`.

When reading this file, consider the [change in isolation](https://github.com/github/cmark-gfm/pull/229/commits/bb117ffa7f0dcccdc4d7773f8585ef07e2402f36) and the [regression test introduced for it](https://github.com/github/cmark-gfm/pull/229/commits/7fa237247371bb3067bcb8fcb4ce74bc95ac04e0).

Reply:
This is safe if `literal->data` is NUL-terminated, but otherwise, we'll need to check the length first.

Reply:
:+1: First, I now check if literal->len has _any_ character.